runtime/binding-http/src/test/java/io/aklivity/zilla/runtime/binding/http/internal/config/PercentEncodableStringComparatorTest.java

```diff
@@ -0,0 +1,286 @@
+        PercentEncodableStringComparator comparator = new PercentEncodableStringComparator();
+
+        // WHEN
+        int result = comparator.compare("hello", "%68%65%6C%6C%6F%49");
+
+        // THEN
+        assertThat(result, lessThan(0));
+    }
+
+    @Test
+    public void shouldBeLessThanZeroWhenComparesShorterAndLongerWhereFirstIsEncoded()
+    {
+        // GIVEN
+        PercentEncodableStringComparator comparator = new PercentEncodableStringComparator();
+
+        // WHEN
+        int result = comparator.compare("%68%65%6C%6C%6F", "hello1");
+
+        // THEN
+        assertThat(result, lessThan(0));
+    }
+
+    @Test
+    public void shouldBeLessThanZeroWhenComparesShorterAndLongerWhereBothAreEncoded()
+    {
+        // GIVEN
+        PercentEncodableStringComparator comparator = new PercentEncodableStringComparator();
+
+        // WHEN
+        int result = comparator.compare("%68%65%6C%6C%6F", "%68%65%6C%6C%6F%49");
+
+        // THEN
+        assertThat(result, lessThan(0));
+    }
+
+    @Test
+    public void shouldBeGreaterThanZeroWhenComparesLargerAndSmaller()
+    {
+        // GIVEN
+        PercentEncodableStringComparator comparator = new PercentEncodableStringComparator();
+
+        // WHEN
+        int result = comparator.compare("hello", "ciao");
+
+        // THEN
+        assertThat(result, greaterThan(0));
+    }
+
+    @Test
+    public void shouldBeGreaterThanZeroWhenComparesLargerAndSmallerWhereSecondIsEncoded()
+    {
+        // GIVEN
+        PercentEncodableStringComparator comparator = new PercentEncodableStringComparator();
+
+        // WHEN
+        int result = comparator.compare("hello", "%63%69%61%6F");
+
+        // THEN
+        assertThat(result, greaterThan(0));
+    }
+
+    @Test
+    public void shouldBeGreaterThanZeroWhenComparesLargerAndSmallerWhereFirstIsEncoded()
+    {
+        // GIVEN
+        PercentEncodableStringComparator comparator = new PercentEncodableStringComparator();
+
+        // WHEN
+        int result = comparator.compare("%68%65%6C%6C%6F", "ciao");
+
+        // THEN
+        assertThat(result, greaterThan(0));
+    }
+
+    @Test
+    public void shouldBeGreaterThanZeroWhenComparesLargerAndSmallerWhereBothAreEncoded()
+    {
+        // GIVEN
+        PercentEncodableStringComparator comparator = new PercentEncodableStringComparator();
+
+        // WHEN
+        int result = comparator.compare("%68%65%6C%6C%6F", "%63%69%61%6F");
+
+        // THEN
+        assertThat(result, greaterThan(0));
+    }
+
+    @Test
+    public void shouldBeGreaterThanZeroWhenComparesLongerAndShorter()
+    {
+        // GIVEN
+        PercentEncodableStringComparator comparator = new PercentEncodableStringComparator();
+
+        // WHEN
+        int result = comparator.compare("hello1", "hello");
+
+        // THEN
+        assertThat(result, greaterThan(0));
+    }
+
+    @Test
+    public void shouldBeGreaterThanZeroWhenComparesLongerAndShorterWhereSecondIsEncoded()
```

Comment:
`shouldCompareLongerUnencodedGreaterThanShorterEncoded`
Let's rename the others to a similar form.

Reply:
OK